### PR TITLE
Viewshed

### LINKED ISSUE #2910: Unexpected Viewshed results
I'm getting unexpected viewshed results.

1. Results are missing land surface that are clearly visible, only flat ocean area is tagged as visible
2. Results are short of the maxDistance=3218.69
3. There's a repeating, no data, pattern in the visible ocean area
gt_viewshed2

```
val point15 = Viewpoint(x = 370815.36, y = 3734806.85, viewHeight = 1.8, angle = 0, fieldOfView = 360, altitude = 0)
val layerVs: TileLayerRDD[SpatialKey] = layerRdd.viewshed(Seq(point15), maxDistance=3218.69, curvature=true)
```

### PR DESCRIPTION
**Pull request by jamesmcclain**
- [x] Artifacts
- [x] Meters per pixel calculation

## Overview

The issue is that some fidelity is lost when visibility information is transferred between adjacent tiles.  The reconstituted data, a collection of line segments associated with each tile, is not guaranteed to cover every pixel in the tile.  The solution presented here is to optionally allow light to scatter perpendicularly from each ray.

~~There have been no changes made to the unit tests because it is unclear how to test a fix like this.  Suggestions welcome.~~  Can be tested with https://github.com/jamesmcclain/ViewshedDebug (locally-publish this branch then type `./sbt "project vs" assembly`, then `$SPARK_HOME/bin/spark-submit --driver-memory=8G --master='local[*]' vs-assembly-0.jar`).

### Artifacts ###

Before:

Screenshot_2019-05-07_19-23-33

Screenshot_2019-05-07_19-23-09

After:

Screenshot_2019-05-07_19-23-25

Screenshot_2019-05-07_19-23-17

### Meters per Pixel ###

Before:

Screenshot_2019-05-08_11-58-33

After:

Screenshot_2019-05-08_11-58-57

Comparison:

Screenshot_2019-05-08_11-58-47

### Checklist

- [x] `docs/CHANGELOG.rst` updated, if necessary
- [x] `docs` guides update, if necessary
- [x] New user API has useful Scaladoc strings
- [x] Unit tests added for bug-fix or new feature

### Demo

Optional. Screenshots/REPL

### Notes

Optional. Ancillary topics, caveats, alternative strategies that didn't work out, anything else.

Closes #2910

### CODE REVIEW

File: raster/src/main/scala/geotrellis/raster/viewshed/R2Viewshed.scala [hunk 1]
```diff
@@ -90,8 +90,12 @@ object R2Viewshed extends Serializable {
     * @param  cols  The number of columns
     * @param  rows  The number of rows
     */
-  def generateEmptyViewshedTile(cols: Int, rows: Int) =
-    ArrayTile.empty(IntConstantNoDataCellType, cols, rows)
+  def generateEmptyViewshedTile(cols: Int, rows: Int, debug: Int = 0) = {
```
Comment (pomadchin) [inline raster/src/main/scala/geotrellis/raster/viewshed/R2Viewshed.scala]: It looks like `debug` should be `Boolean` here, if you still want to keep this flag.

File: docs/CHANGELOG.rst [hunk 2]
```diff
@@ -1,6 +1,14 @@
 Changelog
 =========
 
+3.0.1-SNAPSHOT
```
Comment (pomadchin) [inline docs/CHANGELOG.rst]: Can you move your updated into `3.0.0-SNAPSHOT`, please?

Comment (jamesmcclain) [reply docs/CHANGELOG.rst]: Oh, sure.

**Comment by pomadchin** [comment]
Thanks, James!

**Pull request merged**